### PR TITLE
Corrected color of up arrow in Settings Activity and removed unnecessary…

### LINKED ISSUE #692: Inconsistent Color in Up Arrow in Settings Activity
**Describe the issue**
The up arrow ( Back button ) in Settings Activity has a different color than the rest of the app creating bad User Experience. The color must be white in dark mode and black in dark mode for consistency.

**To Reproduce or Feature details**
1. Click on the settings icon on the action bar.
2. See up arrow being of yellow color.

**Screenshot of error**
The up arrow is yellow in dark mode.

WhatsApp Image 2020-11-05 at 4 26 04 PM

The up arrow is blue in light mode.

WhatsApp Image 2020-11-05 at 4 41 22 PM

**Expected behavior**
The up arrow should be white in dark mode.

WhatsApp Image 2020-11-05 at 4 30 10 PM

and black in light mode.

WhatsApp Image 2020-11-05 at 4 33 18 PM

**Screenshots**
Included above.

**Smartphone (please complete as much of the following information as possible):**
 - Device: Redmi Note 8 Pro
 - OS: Android 10

### PR DESCRIPTION
**Pull request by Prakhar-Agarwal-byte**
…ry files and code.
@chirag-jn @cpg  Please review the Pull Request for the issue - [Inconsistent Color in Up Arrow in Settings Activity ](https://github.com/amahi/android/issues/692) ( #692 ).

### CODE REVIEW

**Comment by chirag-jn** [comment]
We thought about this change before. But the issue was, the tint wasn't visible on Android 5.0 devices and in some cases, caused the app to crash too. Please run the app in an emulator running Android 5.0, 6.0 and 8.1 and see if it works fine for those versions.

File: src/main/res/values/themes.xml [hunk 1]
```diff
@@ -45,7 +45,6 @@
         <item name="android:actionBarStyle">@style/Theme.Amahi.ActionBar</item>
         <item name="android:windowBackground">@color/background_primary</item>
         <item name="android:textColorPrimaryInverse">@color/primary_text_material_light</item>
-        <item name="android:tint">@color/accent</item>
```
Comment (chirag-jn) [inline src/main/res/values/themes.xml:48]: Why did you remove this statement?

File: src/main/res/drawable/ic_settings_upload_photo.xml [hunk 2]
```diff
@@ -4,6 +4,6 @@
     android:viewportWidth="24"
     android:viewportHeight="24">
     <path
-        android:fillColor="#000000"
+        android:fillColor="@color/accent"
```
Comment (chirag-jn) [inline src/main/res/drawable/ic_settings_upload_photo.xml]: Hmm, such references don't work for Android with SDK version less than 21. Since we recently changed the `minSDKVersion` to 21, this shouldn't be an issue. However, keeping `android:tint` overrides this drawable property.

File: src/main/res/values/themes.xml (hunk 1 above)
Comment (prakharagarwal-dev) [reply src/main/res/values/themes.xml:48]: I removed this statement because it was overriding the color of the up button. Instead, I used fillColor property to overcome this situation. I am going to run this in an emulator running Android 5.0, 6.0, and 8.1 and add a screenshot if it works on those versions.

Comment (chirag-jn) [reply src/main/res/values/themes.xml:48]: I feel it's better to keep android:tint property so that uniformity is maintained. In case we plan to change tint in the future, we can change that with a single line change. Further, we won't have to explicitly write fillColor in all drawable files.

**Comment by Prakhar-Agarwal-byte** [comment]
@chirag-jn It is working on all Android 5.0, 6.0, and 8.1 versions. I have put some screenshots, please see it.

**Android 5.0**
Light mode

image

Dark mode

image

**Android 6.0**
Light mode

image

Dark mode

image

**Android 8.1**
Light mode

image

Dark mode

image

File: src/main/res/values/themes.xml (hunk 1 above)
Comment (prakharagarwal-dev) [reply src/main/res/values/themes.xml:48]: What I think is first, we are just using android: tint property just in Settings Activity and nowhere other in the app and It is causing undesirable behavior and second, we can now also change color of all icons through changing just color of accent in colors.xml file. I have attached screenshots in my comment below proving it works on all android versions not less than API 21.

Comment (chirag-jn) [reply src/main/res/values/themes.xml:48]: What do you mean by "undesirable behavior"?

Comment (prakharagarwal-dev) [reply src/main/res/values/themes.xml:48]: I mean that showing different color than expected.

Comment (prakharagarwal-dev) [reply src/main/res/values/themes.xml:48]: @chirag-jn I have added the `android: tint` property back and implemented another way to solve the issue. Please review it.

**Points covered in new implementation:-**
1. Now we will not have to explicitly write fillColor in all drawable files.
2. We can change the tint of all drawables by changing only a single line.
3. I tested the new implementation on Android 5.0, 6.0, and 8.1 and it working correctly in them also. (Screenshots below)


**Android 5.0**
Light mode

image


Dark mode

image



**Android 6.0**
Light mode

image


Dark mode

image


**Android 8.1**
Light mode

image


Dark mode

image

**Comment by cpg** [comment]
Where are we on this PR @chirag-jn ?

**Comment by Prakhar-Agarwal-byte** [comment]
@chirag-jn I think this PR is working fine. Please approve this.

**Review by chirag-jn — APPROVED** [review]
Looks good.
